study/src/org/labkey/study/model/DatasetDomainKind.java

```diff
@@ -314,19 +333,41 @@ public boolean canCreateDefinition(User user, Container container)
         return container.hasPermission(user, AdminPermission.class);
     }
 
+    @Nullable
     @Override
-    public Domain createDomain(GWTDomain domain, JSONObject arguments, Container container, User user,
-        @Nullable TemplateInfo templateInfo)
+    public DatasetDomainKindProperties getDomainKindProperties(@NotNull GWTDomain domain, Container container, User user)
+    {
+//        RP Q: is this the correct approach here? How to guard against these nullpointers?
+        int id = StudyService.get().getDatasetIdByName(container, domain.getName());
+        Dataset ds = StudyService.get().getDataset(container, id);
+        return new DatasetDomainKindProperties(ds);

```

Comment:
@RosalineP looking at the DatasetDomainKind.java class as I have it on develop (without these fb changes), there looks to be a getDatasetDefinition() helper method that you can use for this. So I believe the following should work (since DatasetDefinition class is in implementation of the Dataset interface):
```suggestion
        Dataset ds = getDatasetDefinition(domain.getDomainURI());
        return new DatasetDomainKindProperties(ds);
```